util/headerreader/header_reader.go

```diff
@@ -393,14 +393,21 @@ func headerIndicatesFinalitySupport(header *types.Header) bool {
 	return false
 }
 
+func HeadersEqual(ha, hb *types.Header) bool {
+	if (ha == nil) != (hb == nil) {
+		return false
+	}
+	return (ha == nil) && (hb == nil) || ha.Hash() == hb.Hash()
```

Comment:
Nit: I know `&&` has higher precedence than `||` so this works, but let's add some parens around the && e.g. `(ha == nil && hb == nil)` to clarify it

Reply:
Done.